### PR TITLE
Do not wait until config watcher is ready in order to launch the Atom window

### PR DESCRIPTION
**Pull request by rafeca**
Based on some profiling done locally, this `await` takes ~300-500ms on Windows machines (I couldn't find the root cause of it but may be related to https://github.com/atom/atom/pull/19242).

### CODE REVIEW

**Review by smashwilson — APPROVED** [review]
👍 Good catch.

**Comment by 50Wliu** [comment]
Do you think a comment should be added saying why it's not awaited? Because I could potentially see someone changing it back in the future if they weren't aware.

**Review by nathansobo — COMMENTED** [review]
I'm concerned about this introducing a race condition where we don't catch config file changes that occur just after the window starts up. Does it make sense to skip the `await` only on Windows so that we only introduce the race condition when it is the lesser of two evils? The downside of *that* is that we're introducing inconsistent behavior across platforms.

How can we remember to revisit this change once we upgrade Electron and see if it's still an issue?

**Comment by rafeca** [comment]
> I'm concerned about this introducing a race condition where we don't catch config file changes that occur just after the window starts up. Does it make sense to skip the await only on Windows so that we only introduce the race condition when it is the lesser of two evils? The downside of that is that we're introducing inconsistent behavior across platforms.

I can do that

> How can we remember to revisit this change once we upgrade Electron and see if it's still an issue?

What do you think about what I suggested in https://github.com/atom/atom/pull/19242#issuecomment-488337557 ?

File: src/main-process/atom-application.js [hunk 1]
```diff
@@ -255,7 +261,14 @@ class AtomApplication extends EventEmitter {
...
     if (!this.configFilePromise) {
       this.configFilePromise = this.configFile.watch()
-      this.disposable.add(await this.configFilePromise)
+
+      // TodoElectronIssue: In electron v2 awaiting the watcher causes some delay
+      // in Windows machines, which affects directly the startup time.
+      if (process.platform === 'win32') {
+        this.configFilePromise.then(disposable => this.disposable.add(disposable))
+      } else {
+        this.disposable.add(await this.configFilePromise)
```
Comment (nathansobo) [inline src/main-process/atom-application.js]: The `this.config.onDidChange` calls below this `await` really do need to happen *after* the config file is loaded. That is what is causing #19323. I have a fix in the works and will get it out later today.

Comment (nathansobo) [reply src/main-process/atom-application.js]: Fixed in #19354.

**Pull request merged**